Submissions/Kademgari_Week 3 - Functions and Objects.playground/Contents.swift

```diff
@@ -0,0 +1,223 @@
+//:
+enum Rank: Int {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+}
+let ace = Rank.ace
+let aceRawValue = ace.rawValue
+func rankComparision(r1:Rank, r2:Rank) -> Bool{
+    return r1.rawValue == r2.rawValue
+}
+rankComparision(r1: ace, r2: ace)
+
+//: 7.)
+//:
+//: Add a `color()` method to `Suit` that returns “black” for spades and clubs, and returns “red” for hearts and diamonds.
+//:
+enum Suit {
+    case spades, hearts, diamonds, clubs
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .spades:
+            return "spades"
+        case .hearts:
+            return "hearts"
+        case .diamonds:
+            return "diamonds"
+        case .clubs:
+            return "clubs"
+        }
+    }
+    func color() -> String{
+        switch self{
+        case .spades,.clubs:
+            return "black"
+        case.hearts,.diamonds:
+            return "red"
+        }
+        
+    }
+}
+let hearts = Suit.hearts
+let heartsDescription = hearts.simpleDescription()
+hearts.color()
+Suit.spades.color()
+
+//: 8.)
+//:
+//: Add a third case to `ServerResponse` and to the switch.
+//:
+enum ServerResponse {
+    case result(String, String)
+    case failure(String)
+    case myResponse(String)
+}
+
+let success = ServerResponse.result("6:00 am", "8:09 pm")
+let failure = ServerResponse.failure("Out of cheese.")
+let myResponse = ServerResponse.myResponse("This is my Response")
+
+switch success {
+case let .result(sunrise, sunset):
+    print("Sunrise is at \(sunrise) and sunset is at \(sunset).")
+case let .failure(message):
+    print("Failure...  \(message)")
+case let .myResponse(message):
+    print("MyResponse... \(message)")
+}
+
+//: 9.)
+//:
+//: Write a function that returns an array containing a full deck of cards, with one card of each combination of rank and suit.
+//:
+struct Card {
+    var rank: Rank
+    var suit: Suit
+    func simpleDescription() -> String {
+        return "The \(rank.simpleDescription()) of \(suit.simpleDescription())"
+    }
+}
+let threeOfSpades = Card(rank: .three, suit: .spades)
+let threeOfSpadesDescription = threeOfSpades.simpleDescription()
+func fullDeck() -> [Card] {
+    var deck = Array (repeating: Card(rank: .ace, suit: .spades), count: 52)
+    let suits = [Suit.spades,Suit.hearts,Suit.diamonds,Suit.clubs]
+    var count = 0
+    
+    for i in 1...13{
+        for suit in suits
+        {
+            deck[count + 1] = Card(rank: Rank(rawValue: Rank.RawValue(i))!, suit: suit)
```

Comment:
count is never updated; we have a deck full of the ace of spades; if we include it, the use of [count+1] array accessor will attempt to access array value 52, which is out of range, causing a runtime crash.

the following solution resolves the issue
```Swift
deck[count] = Card(rank: Rank(rawValue: Rank.RawValue(i))!, suit: suit)
count += 1
```